bin/docpub.js

```diff
@@ -10,18 +10,38 @@
 
 const program = require('commander');
 const pkg = require('../package.json');
-const DocpubPipeline = require('../lib/docpub');
+const Docpub = require('../lib/docpub');
 
 program
     .version(pkg.version)
-    .option('-v --verbose', 'verbose output')
-    .option('-p --path <path>', 'path to documents directory (defaults to process directory)');
+    .option('--doc-path <docPath>', 'path to documents directory (defaults to process directory)')
+    .option('--config-path <configPath>', 'path for the config')
+    .option('-v --verbose', 'verbose output');
+
+program.on('--help', () => {
+    console.log('  Overriding config');
+    console.log('    To override any config option use full option path converted to --kebab-case');
+    console.log('');
+
+    console.log('    Examples:');
+    console.log('      docpub --username user@example.com');
+    console.log('      docpub --token abcdefghijklmn12345');
+    console.log('');
+    console.log('    You can also use environment variables converted to snake_case with');
```

Comment:
I prefer to leave this information here. It's may help avoid confusion why you changed something in config, but changes are not reflected.

Also we have pretty small output of help, I don't think it's too much overloaded:
```
  Usage: docpub [options]

  Options:

    -h, --help                  output usage information
    -V, --version               output the version number
    --doc-path <docPath>        path to documents directory (defaults to process directory)
    --config-path <configPath>  path for the config
    -v --verbose                verbose output

  Overriding config
    To override any config option use full option path converted to --kebab-case

    Examples:
      docpub --username user@example.com
      docpub --token abcdefghijklmn12345

    You can also use environment variables converted to snake_case with
    docpub_ prefix

    Examples:
      docpub_url=my-awesome-docs.zendesk.com docpub
      docpub_token=abcdefghijklmn12345 docpub

    If both cli flag and env var are used, cli flag takes precedence
```
If you think help is hard to read, I would rather consider rephrasing it somehow, then hiding this information somewhere in documentation.